tests/ui/consts/const-eval/dangling.rs

```diff
@@ -1,10 +0,0 @@
-use std::mem;
-
-// Make sure we error with the right kind of error on a too large slice.
-const TEST: () = { unsafe {
-    let slice: *const [u8] = mem::transmute((1usize, usize::MAX));
-    let _val = &*slice; //~ ERROR: evaluation of constant value failed
-    //~| slice is bigger than largest supported object
-} };
```

Comment:
This test no longer errors since `*` no longer checks that the place is dereferenceable (and CTFE doesn't do validity checks that would catch the invalid reference).

Reply:
I can add it back as a `-Zextra-const-ub-checks` test.

Reply:
I guess I meant leaving it in as a normal test that now passes, with a note that it could stop passing at any time and that's not a breaking change. But `-Zextra-const-ub-checks` is also good.

Reply:
This also checks that the error is not reported without the flag, so we will notice if behavior changes again.